src/urbit/lib/chess.hoon

```diff
@@ -865,6 +941,28 @@
   ++  draw-claimable  ::  XX no threefold repetition yet
     ^-  ?
     (legal-move [%end %'½–½'])
+  ++  check-repetition
+    |=  [fen-repetition=(map @t @ud) fen=@t]
+    ^-  (map @t @ud)
+    ?.  (~(has by fen-repetition) fen)
+      %=  $
+        fen-repetition  (~(put by fen-repetition) fen 1)
+      ==
+    ?.  (gte 2 (~(got by fen-repetition) fen))
+      %=  $
+        fen-repetition  (~(put by fen-repetition) fen 2)
+      ==
+      %=  $
+        fen-repetition  (~(put by fen-repetition) fen +((~(got by fen-repetition) fen)))
+      ==
```

Comment:
This function doesn't do what it claims it does: it's not returning an updated map, it's returning the `$` arm with a modified subject (which I think actually means that it's looping forever). Aside from that, the name is misleading since it's not "checking" but updating a `map`. Further more, what it does for the app doesn't need it's own function: line `956` is really all that needs to be done to increment the count.